data/src/stream.rs

```diff
@@ -280,12 +280,17 @@ async fn connect(
 
     let (sender, receiver) = mpsc::channel(100);
 
+    let mut client = Client::new(server, config, sender);
+    if let Err(e) = client.connect() {
+        log::error!("Error when connecting client: {:?}", e);
+    }
```

Comment:
You can use "inspect" to log the error without needing to assign and check an intermediate variable

Reply:
I prefer using this construction, since `.inspect_err` returns the original result that needs to be used.

So at best this would be 
```rust
let _ = client.connect().inspect_err(|e| log::error!("Error when connecting client: {:?}", e)));
```
 which I don't think is cleaner